File: db4o.net/Db4objects.Db4o/Db4objects.Db4o/Reflect/Core/PlatformReflectConstructor.cs
/* Copyright (C) 2004 - 2011  Versant Inc.  http://www.db4o.com */

using System;
using Db4objects.Db4o.Internal;
using Db4objects.Db4o.Reflect;
using Db4objects.Db4o.Reflect.Core;

namespace Db4objects.Db4o.Reflect.Core
{
	public class PlatformReflectConstructor : IReflectConstructor
	{
		private static readonly IReflectClass[] ParameterTypes = new IReflectClass[] {  };

		private Type _clazz;

		public PlatformReflectConstructor(Type clazz)
		{
			_clazz = clazz;
		}

		public virtual IReflectClass[] GetParameterTypes()
		{
			return ParameterTypes;
		}

		public virtual object NewInstance(object[] parameters)
		{
			return ReflectPlatform.CreateInstance(_clazz);
		}
	}
}

File: db4oj/core/src/com/db4o/reflect/core/PlatformReflectConstructor.java
package com.db4o.reflect.core;

import com.db4o.internal.*;
import com.db4o.reflect.*;

public class PlatformReflectConstructor implements ReflectConstructor {

	private static final ReflectClass[] PARAMETER_TYPES = new ReflectClass[]{};

	private Class _clazz;
	
	public PlatformReflectConstructor(Class clazz) {
		_clazz = clazz;
	}
	
	public ReflectClass[] getParameterTypes() {
		return PARAMETER_TYPES;
	}

	public Object newInstance(Object[] parameters) {
		return ReflectPlatform.createInstance(_clazz);
	}

}
